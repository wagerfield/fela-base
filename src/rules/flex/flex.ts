export interface FlexRuleOptions {
  grow?: any
  shrink?: any
  basis?: any
  align?: any
  justify?: any
}

export default ({
  grow,
  shrink,
  basis,
  align,
  justify
}: FlexRuleOptions = {}) => ({
  flexGrow: grow,
  flexShrink: shrink,
  flexBasis: basis,
  alignSelf: align,
  justifySelf: justify
})
