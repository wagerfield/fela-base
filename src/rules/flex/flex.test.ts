import flex from "./flex"

describe("flex(props)", () => {
  it("renders expected snapshots", () => {
    expect(flex()).toMatchSnapshot()
    expect(
      flex({
        grow: 1
      })
    ).toMatchSnapshot()
  })
})
