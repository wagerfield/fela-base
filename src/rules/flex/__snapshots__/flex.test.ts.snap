// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flex(props) renders expected snapshots 1`] = `
Object {
  "alignSelf": undefined,
  "flexBasis": undefined,
  "flexGrow": undefined,
  "flexShrink": undefined,
  "justifySelf": undefined,
}
`;

exports[`flex(props) renders expected snapshots 2`] = `
Object {
  "alignSelf": undefined,
  "flexBasis": undefined,
  "flexGrow": 1,
  "flexShrink": undefined,
  "justifySelf": undefined,
}
`;
